src/video/video.js

```diff
@@ -113,15 +123,26 @@ export default (
 
         componentDidUpdate(lastProps) {
             if (!this.props.videoId) return;
-            if (lastProps.videoId !== this.props.videoId) {
+
+            const hasIdChanged = lastProps.videoId !== this.props.videoId;
+            const videoElIdDoesNotMatchVideoId =
+                this.videoEl && this.videoEl.id !== this.props.videoId;
+
+            if (hasIdChanged) {
+                this.unbindEvents();
+                this.getElementAndUpdateState(this.props.videoId);
+            } else if (videoElIdDoesNotMatchVideoId) {
```

Comment:
could this be `if (hasIdChanged || videoElIdDoesNotMatchVideoId) { ... }`

Reply:
🤦 doh yes